scripts/BadgeManager.js

```diff
@@ -2,20 +2,44 @@
 
 'use strict';
 
-function BadgeManager(addEventListener) {
+function BadgeManager(addEventListener, isEnabled) {
 
 	addEventListener("hourMusic", function(hour) {
-		chrome.browserAction.setBadgeText({ text: formatHour(hour) });
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: formatHour(hour) });
+		}
+		play(false);
 	});
 
 	addEventListener("kkStart", function() {
-		chrome.browserAction.setBadgeText({ text: "KK" });
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: "KK" });
+		}
+		play(true);
 	});
 
 	addEventListener("pause", function() {
 		chrome.browserAction.setBadgeText({ text: "" });
+		pause();
 	});
+	
+	function play(kk) {
```

Comment:
I'm not the biggest fan of the functions being called "play" and "pause", since that makes them sound like they're controlling the music. Maybe have one function called changeIcon that takes two variables - isPlaying and isKK?

This is just a minor nitpick though.